src/panels/MissionsPanel/TimeLine/Rectangle.tsx

```diff
@@ -0,0 +1,80 @@
+import { Text, Tooltip } from '@mantine/core';
+import { ScaleLinear, ScaleTime } from 'd3';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+import { useAppSelector } from '@/redux/hooks';
+import { DisplayType } from '@/types/enums';
+import { Phase } from '@/types/mission-types';
+
+import { DisplayedPhase } from '../MissionContent';
+import { jumpToTime } from '../util';
+
+interface RectangleProps {
+  scale: number; // d3 scale 'k' value
+  xScale: ScaleLinear<number, number, never>;
+  yScale: ScaleTime<number, number, never>;
+  nestedLevels: number;
+  setDisplayedPhase: (phase: DisplayedPhase) => void;
+  phase: Phase;
+  nestedLevel: number;
+  padding?: number;
+  color?: React.CSSProperties['color'];
+}
+
+export function Rectangle({
+  scale,
+  xScale,
+  yScale,
+  nestedLevels,
+  setDisplayedPhase,
+  phase,
+  nestedLevel,
+  padding = 0,
+  color
+}: RectangleProps) {
+  const now = useAppSelector((state) => state.time.timeCapped);
```

Comment:
See the same comment as before. Would be creat to not have to depend on `now` in all the components. 

Here it seems like it might still be needed, however? Although, to avoid constant rerendering you could return `isCurrent` instead: 
```suggestion
  const isCurrent = useAppSelector((state) => {
    const now = state.time.timeCapped;
    const isBeforeEndTime = now < endTime.valueOf();
    const isAfterBeginning = now > startTime.valueOf();
    return isBeforeEndTime && isAfterBeginning;
  });
```
Of that would mean moving the variables for start and end time to before this selector

Reply:
I'm on board with this, however, the whole timeline is being rerendered every second since the `MissionContent` is also subscribed to time. We should perhaps go back to this component later and see if we can avoid the constant rerenders?